### PR TITLE
some improvements from Xiaomi team

### PR DESCRIPTION
**Pull request by qinzuoyan**

### CODE REVIEW

File: src/replica_server/replication_lib/mutation.cpp [hunk 1]
```diff
@@ -285,7 +286,18 @@ mutation_queue::mutation_queue(gpid gpid, int max_concurrent_op /*= 2*/, bool ba
 
 mutation_ptr mutation_queue::add_work(task_code code, dsn_message_t request, replica* r)
 {
-    // batch and add to work queue
+    task_spec* spec = task_spec::get(code);
+
+    // if not allow write batch, switch work queue
+    if (_pending_mutation && !spec->rpc_request_is_write_allow_batch)
```
Comment (qinzuoyan) [inline src/replica_server/replication_lib/mutation.cpp]: but it is task-code level option. If I put it in replication-specific option, maybe as:
```
[replication]
rpc_request_disable_batch_write = RPC_RRDB_RRDB_MULTI_PUT,RPC_RRDB_RRDB_INCR
```
or do you have better suggestion?

Comment (imzhenyu) [inline src/replica_server/replication_lib/mutation.cpp]: This is doable. Or, you can have your own options of "task-spec" array, as we have done in cases such as [profiler](https://github.com/Microsoft/rDSN/blob/master/src/plugins/tools.common/profiler.cpp#L86). In that way, we avoid changing the core code due to the requirement from certain plugins.

Comment (qinzuoyan) [inline src/replica_server/replication_lib/mutation.cpp]: it is a time-consuming work to fix it, or we can leave a issue to the future, not blocking the open source work of pegasus?

Comment (imzhenyu) [inline src/replica_server/replication_lib/mutation.cpp]: of course.

**Pull request merged**